### PR TITLE
feat: remove containerd service dependency

### LINKED ISSUE #137: Feature request: remove containerd service dependency

### PR DESCRIPTION
**Pull request by Desiki-high**
Acceld depend on local containerd service, we should replace it by create content store by containerd pkg.

We use [github.com/containerd/containerd/content/local](https://github.com/containerd/containerd/tree/main/content/local) to remove containerd service dependency.

Break Change:
1. remove containerd config.
2. add work_dir in provider for content store work.

resolve #137.

### CODE REVIEW

File: pkg/adapter/adapter.go [hunk 1]
```diff
@@ -145,16 +127,6 @@ func (adp *LocalAdapter) Dispatch(ctx context.Context, ref string, sync bool) er
...
-	msg := "containerd service is unhealthy"
-	if err != nil {
-		return errors.Wrap(err, msg)
-	}
-
-	if !health {
-		return fmt.Errorf(msg)
-	}
-
+	//TODO：return the status of boltdb
```
Comment (imeoer) [inline pkg/adapter/adapter.go:130]: `// TODO`

File: pkg/config/config.go [hunk 2]
```diff
@@ -46,8 +46,8 @@ type MetricConfig struct {
 }
 
 type ProviderConfig struct {
-	Source     map[string]SourceConfig `yaml:"source"`
-	Containerd ContainerdConfig        `yaml:"containerd"`
+	Source  map[string]SourceConfig `yaml:"source"`
+	WorkDir string                  `yaml:"work_dir"`
```
Comment (imeoer) [inline pkg/config/config.go:50]: Should also remove the `ContainerdConfig` structure definition.

File: pkg/content/local.go [hunk 3]
```diff
@@ -95,25 +94,23 @@ func (pvd *LocalProvider) Push(ctx context.Context, desc ocispec.Descriptor, ref
...
-		containerd.WithResolver(resolver),
-		containerd.WithPlatformMatcher(pvd.platformMC),
+	rc := &containerd.RemoteContext{
+		Resolver:        resolver,
+		PlatformMatcher: pvd.platformMC,
 	}
 
-	// TODO: sets max concurrent uploaded layer limit by containerd.WithMaxConcurrentUploadedLayers.
-	return pvd.client.Push(ctx, ref, desc, opts...)
+	return push(ctx, pvd.store, rc, desc, ref)
```
Comment (imeoer) [inline pkg/content/local.go:102]: Still need to keep the TODO about `WithMaxConcurrentUploadedLayers`?

File: pkg/content/local.go [hunk 4]
```diff
@@ -67,25 +73,18 @@ func (pvd *LocalProvider) Pull(ctx context.Context, ref string) error {
...
 
-	// Pull the source image from remote registry.
-	_, err = pvd.client.Fetch(ctx, ref, opts...)
+	img, err := fetch(ctx, pvd.store, rc, ref, 0)
 	if err != nil {
 		return errors.Wrap(err, "pull source image")
 	}
+	pvd.mutex.Lock()
+	defer pvd.mutex.Unlock()
+	pvd.images[ref] = &img.Target
```
Comment (imeoer) [inline pkg/content/local.go:87]:
```
pvd.mutex.Lock()
defer pvd.mutex.Unlock()
pvd.images[ref] = &img.Target
```

Can we wrap this into a `set` method?

File: pkg/content/local.go [hunk 5]
```diff
@@ -95,25 +94,23 @@ func (pvd *LocalProvider) Push(ctx context.Context, desc ocispec.Descriptor, ref
...
+	return push(ctx, pvd.store, rc, desc, ref)
 }
 
 func (pvd *LocalProvider) Image(ctx context.Context, ref string) (*ocispec.Descriptor, error) {
-	image, err := pvd.client.GetImage(ctx, ref)
-	if err != nil {
-		return nil, err
+	pvd.mutex.Lock()
+	defer pvd.mutex.Unlock()
+	if desc, ok := pvd.images[ref]; ok {
```
Comment (imeoer) [inline pkg/content/local.go:108]: Ditto, wrap into a `get` method.

File: pkg/adapter/adapter.go (hunk 1 above)
Comment (Desiki-high) [reply pkg/adapter/adapter.go:130]: fixed

File: pkg/config/config.go (hunk 2 above)
Comment (Desiki-high) [reply pkg/config/config.go:50]: fixed

File: pkg/content/local.go [hunk 6]
```diff
@@ -96,24 +94,33 @@ func (pvd *LocalProvider) Push(ctx context.Context, desc ocispec.Descriptor, ref
...
-	return &target, nil
+	return pvd.getImages(ref)
 }
 
 func (pvd *LocalProvider) ContentStore() content.Store {
-	return pvd.client.ContentStore()
+	return pvd.store
+}
+
+func (pvd *LocalProvider) setImages(ref string, image *ocispec.Descriptor) {
```
Comment (imeoer) [inline pkg/content/local.go:113]: `setImage` ?

File: pkg/content/local.go [hunk 7]
```diff
@@ -96,24 +94,33 @@ func (pvd *LocalProvider) Push(ctx context.Context, desc ocispec.Descriptor, ref
...
+	pvd.mutex.Lock()
+	defer pvd.mutex.Unlock()
+	pvd.images[ref] = image
+}
+
+func (pvd *LocalProvider) getImages(ref string) (*ocispec.Descriptor, error) {
```
Comment (imeoer) [inline pkg/content/local.go:119]: `getImage` ?

Comment (Desiki-high) [reply pkg/content/local.go]: fixed

**Review by imeoer — APPROVED** [review]

**Pull request merged**